dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx

```diff
@@ -3,102 +3,487 @@
+        </CustomOverlay>
+      )}
+    </>
+  );
+};
 
-  const renderCellValue = useMemo(() => {
-    return ({ rowIndex, columnId }) => {
-      let adjustedRowIndex = rowIndex;
+const CustomOverlay = ({ children }: { children: any }) => {
+  return (
+    <div
+      className={
+        uiSettingsService.get('theme:darkMode')
+          ? 'custom-overlay custom-overlay-dark'
+          : 'custom-overlay custom-overlay-light'
+      }
+    >
+      {children}
+    </div>
+  );
+};
 
-      // If we are doing the pagination (instead of leaving that to the grid)
-      // then the row index must be adjusted as `data` has already been pruned to the page size
-      adjustedRowIndex = rowIndex - pagination.pageIndex * pagination.pageSize;
+const GridHeader = ({
+  isFullScreen,
+  setIsFullScreenHandler,
+  selectedRowDensity,
+  selectDensityHandler,
+  columnVisiblityHandler,
+  columns,
+  columnVisibility,
+}: {
+  isFullScreen: boolean;
+  setIsFullScreenHandler: (v: boolean) => void;
+  selectedRowDensity: any;
+  selectDensityHandler: (v: any) => void;
+  columnVisiblityHandler: (visible: boolean, feild: any) => void;
+  columns: any;
+  columnVisibility: any;
+}) => {
+  return (
+    <>
+      <EuiFlexGroup
+        responsive={true}
+        gutterSize="none"
+        justifyContent="flexStart"
+        style={{ position: 'relative' }}
+      >
+        <EuiFlexItem style={{ maxWidth: '150px' }}>
+          <DensityPopover
+            selectedDensity={selectedRowDensity}
+            selectDensityHandler={selectDensityHandler}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem style={{ maxWidth: '150px' }}>
+          <ColumnVisiblityPopover
+            columnVisiblityHandler={columnVisiblityHandler}
+            columns={columns}
+            columnVisibility={columnVisibility}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem style={{ maxWidth: '150px' }}>
+          <EuiButtonEmpty
+            iconSize="s"
+            color="text"
+            aria-label="Next"
+            iconType="fullScreen"
+            onClick={() => setIsFullScreenHandler(true)}
+          >
+            Full screen
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+        {isFullScreen && (
+          <EuiIcon
+            type="cross"
+            onClick={() => setIsFullScreenHandler(false)}
+            style={{ position: 'absolute', right: 20, cursor: 'pointer', top: 20 }}
+          />
+        )}
+      </EuiFlexGroup>
+    </>
+  );
+};
 
-      return data.hasOwnProperty(adjustedRowIndex) ? data[adjustedRowIndex][columnId] : null;
-    };
-  }, [data, pagination.pageIndex, pagination.pageSize]);
+const GridFooter = ({
+  onPageSizeChanged,
+  pageSize,
+  activePage,
+  goToPage,
+  pageCount,
+}: {
+  onPageSizeChanged: (val: number) => void;
+  goToPage: (val: number) => void;
+  pageSize: number;
+  activePage: number;
+  pageCount: number;
+}) => {
+  return (
+    <EuiFlexGroup
+      responsive={true}
```

Comment:
Same as mentioned before. You can just say `responsive`. The value `true` is implicit in that case. Check other places too. I saw many instances of this kind of code.

Reply:
okay